### ISSUE TITLE
Pipeable concatAll and mergeAll types broken

### ISSUE THREAD

**Issue by AustinMatherne** [ReactiveX/rxjs#3290]
**RxJS version:**
5.5.6
**Code to reproduce:**
```ts
import { Observable } from 'rxjs/Observable';
import { of } from 'rxjs/observable/of';
import 'rxjs/add/operator/concatAll';
import { concatAll } from 'rxjs/operators/concatAll';
import 'rxjs/add/operator/mergeAll';
import { mergeAll } from 'rxjs/operators/mergeAll';
import 'rxjs/add/operator/switch';
import { switchAll } from 'rxjs/operators/switchAll';

const prototypeConcatAll: Observable<number> = of(of(1)).concatAll();
const pipeableConcatAll: Observable<number> = of(of(1)).pipe(concatAll()); // Observable<Observable<number>> is not assignable to type Observable<number>

const prototypeMergeAll: Observable<number> = of(of(1)).mergeAll();
const pipeableMergeAll: Observable<number> = of(of(1)).pipe(mergeAll());// Observable<Observable<number>> is not assignable to type Observable<number>

const prototypeSwitchAll: Observable<number> = of(of(1)).switch();
const pipeableSwitchAll: Observable<number> = of(of(1)).pipe(switchAll());
```
**Expected behavior:**
`concatAll` and `mergeAll` pipeable operators transform type `Observable<T>` > `T` the same as the prototype operators.

**Actual behavior:**
`concatAll` and `mergeAll` pass types through `Observable<T>` > `Observable<t>`.

**Additional information:**
While `concatAll` and `mergeAll`are misbehaving, `switchAll` is working as expected.

**Comment by martinsik** [comment]
I think this is already covered here #2759.

**Comment by AustinMatherne** [comment]
Thanks for the link. That bug is similar, but appears to be different.

This one only applies to pipeable operators. The prototype operators work just fine.
More importantly, this bug is still present in rxjs@6.0.0-alpha.2.

```ts
import { Observable } from 'rxjs';
import { of } from 'rxjs/create';
import { map, concatAll, mergeAll, switchAll } from 'rxjs/operators';

const concatenated: Observable<number> = of(1).pipe(map(x => of(x)), concatAll());
// Observable<number | Observable<number>> is not assignable to type Observable<number>

const merged: Observable<number> = of(1).pipe(map(x => of(x)), mergeAll());
// Observable<number | Observable<number>> is not assignable to type Observable<number>

const switched: Observable<number> = of(1).pipe(map(x => of(x)), switchAll());

```

**Comment by AustinMatherne** [comment]
It appears [concatAll](https://github.com/ReactiveX/rxjs/blob/a24146f1ad09acd62ea92ee0f8d983d5ab5cf835/src/internal/operators/concatAll.ts#L53-L55) and [mergeAll](https://github.com/ReactiveX/rxjs/blob/a24146f1ad09acd62ea92ee0f8d983d5ab5cf835/src/internal/operators/mergeAll.ts#L51-L53) should be typed as `<T>(): OperatorFunction<Observable<T>, T>`.
I'm happy to open up a PR if this is welcomed.

```ts
export function concatAll<T>(): OperatorFunction<Observable<T>, T> {
  return mergeAll(1);
}
```

In the meantime I'm overriding the types locally
```ts
import {concatAll as rxConcatAll} from 'rxjs/operators/concatAll';
import {mergeAll as rxMergeAll} from 'rxjs/operators/mergeAll';

export const concatAll: <T>() => OperatorFunction<Observable<T>, T> = <any>rxConcatAll;
export const mergeAll: <T>() => OperatorFunction<Observable<T>, T> = <any>rxMergeAll;
```

**Comment by mlegenhausen** [comment]
```ts
declare module 'rxjs/operators/mergeAll' {
  import { UnaryFunction } from 'rxjs/interfaces';
  import { Observable } from 'rxjs/Observable';
  export function mergeAll<T>(concurrent?: number): UnaryFunction<Observable<T>, T>;
}
```

This can be added to a `typings.d.ts` if you just want to the types to be fixed instead of reexporting the whole function. The same should work for `concatAll`.

**Comment by benlesh** [comment]
closed by #3321

**Comment by princemaple** [comment]
Will there be a fix released soon?

**Comment by benlesh** [comment]
This is fixed in 6.0.0.

**Comment by Ammonix** [comment]
@benlesh 
Theoretically this was fixed in https://github.com/ReactiveX/rxjs/commit/f7e4c02444adc98732c2040201337b40e72dfabf#diff-9cc891b15c60df5468b6b9879f1a230aR54 
but for some reason later reverted in https://github.com/ReactiveX/rxjs/commit/d29324501921f56b41f3b43481b30f184fad97c1#diff-9cc891b15c60df5468b6b9879f1a230aR52 
So the bug still exists.
Please reopen this issue.
